kani-driver/src/cbmc_output_parser.rs

```diff
@@ -431,6 +447,26 @@ impl<'a, 'b> Parser<'a, 'b> {
         }
         None
     }
+
+    /// Adds a single line to the CBMC output buffered writer.
+    pub fn add_line_to_buf_writer(&mut self, line: &str) {
+        if let Some(buf_writer) = self.cbmc_out_buf_writer.as_mut() {
+            writeln!(buf_writer, "{}", line).expect(&format!(
```

Comment:
I think `line` already include newlines. Have you checked you're not writing empty newlines for each line? If that's the case, replace with `write!`.

Reply:
Ah, nice catch! Done